src/main/java/Duke/Parser.java

```diff
@@ -0,0 +1,55 @@
+package Duke;
+
+import java.util.Scanner;
+
+/**
+ * This class encapsulates the understanding of commands.
+ * It calls the relevant command depending on the user's input.
+ */
+public class Parser {
+
+    private Scanner sc = new Scanner(System.in);
+
+    /**
+     * Gets the user's input and return the relevant command to execute.
+     * @return A command which determines how the program should react.
+     */
+    public ICommand getInput() {
+        String input = sc.nextLine();
+
+        // Exit
+        if (input.equals("bye")) {
+            return new ByeCommand();
+
+        // List out tasks
+        } else if (input.equals("list")) {
+            return new ListTasksCommand();
+
+        } else if (input.split(" ")[0].equals("done")) {
```

Comment:
Should `input.split(" ")[0]` be extracted out and assigned to a constant? This is so that the program does not have to carry out the split function every time it evaluates a conditional expression.